src/int.cr

```diff
@@ -2776,3 +2776,53 @@ struct UInt128
     self
   end
 end
+
+# Returns a number for given digits and base.
+# The digits are expected as an Enumerable with the least significant digit as the first element.
+#
+# Base must not be less than 2.
+#
+# All digits must be within 0...base.
+#
+# ```
+# Int32.from_digits([5, 4, 3, 2, 1])           # => 12345
+# Int32.from_digits([4, 6, 6, 0, 5], base: 7)  # => 12345
+# Int32.from_digits([45, 23, 1], base: 100)    # => 12345
+#
+# Int32.from_digits([1], base: -2) # => ArgumentError
+# Int32.from_digits([-1]) # => ArgumentError
+# Int32.from_digits([3], base: 2) # => ArgumentError
+# ```
+{% for type in %w(Int8 Int16 Int32 Int64 Int128 UInt8 UInt16 UInt32 UInt64 UInt128) %}
+  def {{type.id}}.from_digits(digits : Enumerable(Int), base : Int = 10) : self
+    if base < 2
+      raise ArgumentError.new("Invalid base #{base}")
+    end
+
+    num : {{type.id}} = 0
+    multiplier : {{type.id}} = 1
+    first_element = true
+
+    digits.each_with_index do |digit, i|
```

Comment:
this `i` is unused; you only need either this or `first_element` to check for the last (or first) element

Reply:
Ah, right.

I tried first to skip the last element. But in that case we need to call `Enumerable#size` that means iterating the whole collection in general case. So the current approach with `first_element` seems to me slightly better from the performance point of view (in general case). Does it make sense?